File: sdkwork-ai-sdk-csharp/Api/Context2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Context2Api
    {
        private readonly HttpClient _client;

        public Context2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Chat with context
        /// </summary>
        public async Task<ContextResponse?> ChatWithAsync(string contextId, ChatWithContextPostRequest body)
        {
            return await _client.PostAsync<ContextResponse>(ApiPaths.AiPath($"/v1/context/{contextId}/chat/completions"), body);
        }

        /// <summary>
        /// Create context
        /// </summary>
        public async Task<ContextResponse?> CreateContextAsync(ContextCreateRequest body)
        {
            return await _client.PostAsync<ContextResponse>(ApiPaths.AiPath("/v1/context/create"), body);
        }

        /// <summary>
        /// Get context
        /// </summary>
        public async Task<ContextResponse?> GetContextAsync(string contextId)
        {
            return await _client.GetAsync<ContextResponse>(ApiPaths.AiPath($"/v1/context/{contextId}"));
        }

        /// <summary>
        /// Delete context
        /// </summary>
        public async Task DeleteContextAsync(string contextId)
        {
            await _client.DeleteAsync<object>(ApiPaths.AiPath($"/v1/context/{contextId}"));
        }
    }
}

File: sdkwork-ai-sdk-swift/Sources/API/Context2Api.swift
import Foundation

public class Context2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Chat with context
    public func chatWith(contextId: String, body: ChatWithContextPostRequest) async throws -> ContextResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/context/\(contextId)/chat/completions"), body: body)
        return response as? ContextResponse
    }

    /// Create context
    public func createContext(body: ContextCreateRequest) async throws -> ContextResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/context/create"), body: body)
        return response as? ContextResponse
    }

    /// Get context
    public func getContext(contextId: String) async throws -> ContextResponse? {
        let response = try await client.get(ApiPaths.aiPath("/v1/context/\(contextId)"))
        return response as? ContextResponse
    }

    /// Delete context
    public func deleteContext(contextId: String) async throws -> Void {
        _ = try await client.delete(ApiPaths.aiPath("/v1/context/\(contextId)"))
    }
}
